frontendUiRootImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/VcsRootComboBox.java

```diff
@@ -45,14 +70,20 @@ public void updateRepositories(@MinLen(1) List<GitRepository> repositories) {
       getModel().update(repositories.asJavaMutable());
     }
 
+    this.setVisible(getModel().getItems().size() > 1);
+
     boolean selectedItemUpdateRequired = selected == null || !getModel().getItems().contains(selected);
-    if (selectedItemUpdateRequired) {
-      getModel().setSelectedItem(repositories.get(0));
+    if (repositories.isEmpty()) {
+      LOG.debug(() -> "No VCS roots found");
+      observers.forEach(o -> o.onSelectionChanged());
+    } else if (selectedItemUpdateRequired) {
+      LOG.debug(() -> "Selecting first VCS root");
+      setSelectedItem(repositories.get(0));
     } else {
+      LOG.debug(() -> "Selecting previously selected VCS root");
+      // VcsRootComboBox#setSelectedItem is omitted to avoid unnecessary observers call
       getModel().setSelectedItem(selected);
```

Comment:
Hmm why `setSelectedItem` in some cases and `getModel().setSelectedItem` in others?

Reply:
The difference is following:
- `getModel().setSelectedItem(...)` does not call observers 
- `setSelectedItem(...)` calls observers

Omitting observers is desirable in the case when selection does not change.